core/file.go

```diff
@@ -168,49 +168,59 @@ func (file *File) Contents(ctx context.Context, offset, limit *int) ([]byte, err
 		return nil, nil
 	}
 
-	r, err := file.Open(ctx)
-	if err != nil {
-		return nil, err
-	}
-	defer r.Close()
```

Comment:
As a follow-up, everywhere that uses this should probably be refactored to use `execInMount`.

It would be nice to remove `file.Open` entirely?

But yeah, this is a nice change, as I read it, the previous implementation actually does a mount/unmount for *every* call to `Read`? Oof.